### PR TITLE
Commits to go above original the repo split

### PR DESCRIPTION
**Pull request by KKoukiou**
This repository was created with:
`git filter-repo --subdirectory-filter ui/webui`
In order to bring back translations, making build system work, setup CI, packit, adjust docs with the new paths and much more, all these commits are needed on top.

Proof that CI works. https://github.com/rhinstaller/anaconda-webui/pull/23

### CODE REVIEW

File: packaging/anaconda-webui.spec.in [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+BuildRequires:  libappstream-glib
+BuildRequires:  make
+BuildRequires: gettext
+
+%define anacondacorever 40.9
+%define cockpitver 275
+
+Requires: cockpit-bridge >= %{cockpitver}
+Requires: cockpit-ws >= %{cockpitver}
+Requires: anaconda-core  >= %{anacondacorever}
```
Comment (KKoukiou) [inline packaging/anaconda-webui.spec.in:36]: I need a requires on firefox here.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        run: git config --global --add safe.directory /__w/anaconda-webui/anaconda-webui
+
+      - name: Workaround for https://github.com/actions/checkout/pull/697
+        run: git fetch --force origin $(git describe --tags):refs/tags/$(git describe --tags)
+
+      - name: Build release
+        run: make dist
+
+      - name: Publish GitHub release
+        uses: cockpit-project/action-release@88d994da62d1451c7073e26748c18413fcdf46e9
```
Comment (jkonecny12) [inline .github/workflows/release.yml:33]: I like this, however, couldn't it be versioned as other actions?

File: packit.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    owner: "@rhinstaller"
+    project: "Anaconda-webui"
+    preserve_project: True
+    actions:
+      post-upstream-clone: make anaconda-webui.spec
+      # HACK: tarball for releases (copr_build, koji, etc.), copying spec's Source0; this
+      # really should be the default, see https://github.com/packit/packit-service/issues/1505
+      create-archive:
+        - sh -exc "curl -L -O https://github.com/rhinstaller/anaconda-webui/releases/download/${PACKIT_PROJECT_VERSION}/${PACKIT_PROJECT_NAME_VERSION}.tar.xz"
+        - sh -exc "ls ${PACKIT_PROJECT_NAME_VERSION}.tar.xz"
```
Comment (jkonecny12) [inline packit.yaml:35]: I think we want to have nightly builds from HEAD?

File: packaging/anaconda-webui.spec.in [hunk 4]
```diff
@@ -0,0 +1,85 @@
+#
+# Copyright (C) 2021 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
```
Comment (rvykydal) [inline packaging/anaconda-webui.spec.in:4]: "Cockpit"?

File: README.rst [hunk 5]
```diff
@@ -9,34 +9,38 @@ Here's where to get the code::
...
+Automated release
+-----------------
 
-For developer convenience, the ESLint can be started explicitly by::
+The intention is that the only manual step for releasing a project is to create
+a signed tag for the version number, which includes a summary of the noteworthy
+changes::
 
-    npm run eslint
+    123
```
Comment (jkonecny12) [inline README.rst:25]: Is this a typo?

**Review by jkonecny12 — COMMENTED** [review]
I have a few notes for fixing. In overall I really like it.

File: .github/workflows/dependabot.yml [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        run: |
+          # Dependabot prefixes the commit with [no-test] which we don't want to keep in the commit
+          title=$(git show --pretty="%s" -s | sed -E "s/\[no-test\]\W+ //")
+          body=$(git show -s --pretty="%b")
+          git commit --amend -m "${title}" -m "${body}" --no-edit node_modules
+          eval $(ssh-agent)
+          ssh-add - <<< '${{ secrets.SELF_DEPLOY_KEY }}'
+          git push --force 'git@github.com:${{ github.repository }}' '${{ github.head_ref }}'
+          ssh-add -D
+          ssh-agent -k
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply .github/workflows/dependabot.yml:82]: These workflows are alligned between all cockpit projects. https://github.com/cockpit-project/cockpit-machines/blob/main/.github/workflows/dependabot.yml 

I can make changes and sync to all of them, but I would rather not diverge, for easing the comparison in the future.

File: Makefile [hunk 7]
```diff
@@ -39,6 +39,36 @@ TEST_LIVE_OS=fedora-rawhide-live-boot
...
+
+po/$(PACKAGE_NAME).metainfo.pot: $(APPSTREAMFILE)
+	xgettext --default-domain=$(PACKAGE_NAME) --output=$@ $<
+
+po/$(PACKAGE_NAME).pot: po/$(PACKAGE_NAME).html.pot po/$(PACKAGE_NAME).js.pot po/$(PACKAGE_NAME).manifest.pot po/$(PACKAGE_NAME).metainfo.pot
+	msgcat --sort-output --output-file=$@ $^
+
+po/LINGUAS:
+	echo $(LINGUAS) | tr ' ' '\n' > $@
+
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply Makefile:71]: Sure - can we clean this up in a followup?

File: .github/workflows/release.yml (hunk 2 above)
Comment (KKoukiou) [reply .github/workflows/release.yml:33]: It's custom action from cockpit. http://git.app.eng.bos.redhat.com/git/cockpit-release.git/

would be fine to use tags, but cockpit uses SHA as well, so I bet they don't tag this.

File: README.rst (hunk 5 above)
Comment (KKoukiou) [reply README.rst:25]: Yes this should not be there, sorry

File: packit.yaml (hunk 3 above)
Comment (KKoukiou) [reply packit.yaml:35]: Sure, we can extend packit configuration with anything extra as a followup.

File: scripts/makeupdates [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+    os.system("find . | cpio -c -o | gzip -9cv > %s/updates.img" % (cwd,))
+    sys.stdout.write("updates.img ready\n")
+
+def usage(cmd):
+    sys.stdout.write("Usage: %s [OPTION]...\n" % (cmd,))
+    sys.stdout.write("Options:\n")
+    sys.stdout.write("    -k, --keep       Do not delete updates subdirectory.\n")
+    sys.stdout.write("    -h, --help       Display this help and exit.\n")
+    sys.stdout.write("    -t, --tag        Make image from TAG to HEAD.\n")
+    sys.stdout.write("    -o, --offset     Make image from (latest_tag - OFFSET) to HEAD.\n")
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply scripts/makeupdates:107]: I dont need most of this script. It's a direct copy from blivet. I did not take a time to clear this out, but I created a Jira for us to replace it with some common script accross repos. https://issues.redhat.com/browse/INSTALLER-3755

File: scripts/makeupdates [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Author: David Cantrell <dcantrell@redhat.com>
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply scripts/makeupdates:21]: Well I really copied it from the blivet repo and this author is there.

File: .github/workflows/weblate-sync-pot.yml [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+          repository: ${{ github.repository }}-l10n
+          ssh-key: ${{ secrets.DEPLOY_KEY }}
+
+      - name: Commit .pot to weblate repo
+        run: |
+          cp src/po/anaconda-webui.pot weblate/anaconda-webui.pot
+          git config --global user.name "GitHub Workflow"
+          git config --global user.email "cockpituous@cockpit-project.org"
+          git -C weblate commit -m "Update source file" -- anaconda-webui.pot
+          git -C weblate push
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply .github/workflows/weblate-sync-pot.yml:42]: indeed this looks better. Since it works, and I don't want to re-test everything now, can we adjust in normal PR after the split happens.

File: README.rst (hunk 5 above)
Comment (KKoukiou) [reply README.rst:25]: Nope. This is the tag message expected structure. THe eslint part moved in to HACKING.rst.

File: .github/workflows/codeql.yml [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language:
+          - javascript
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply .github/workflows/codeql.yml:17]: This file is a direct copy from https://github.com/rhinstaller/anaconda/blob/master/.github/workflows/codeql.yml

I guess we can simplify it but let it be a followup.

**Review by rvykydal — COMMENTED** [review]
I don't have more to add to Jirka. The only thing I didn't look at much is the `makeupdates` script which is already being discussed (and could be improved separately I guess)

File: .github/workflows/dependabot.yml [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+          TITLE: '${{ github.event.pull_request.title }}'
+        with:
+          script: |
+            const title = process.env['TITLE'].replace(/\[no-test\]\W+ /, '')
+            await github.rest.pulls.update({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.issue.number,
+              title,
+            });
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply .github/workflows/dependabot.yml]: Removed. This was incorrectly copied from another cockpit plugin, tht deals with node_modules different than us. I removed it, we will need it in the future, but not now.

File: Makefile [hunk 13]
```diff
@@ -18,11 +18,19 @@
 # one example file in dist/ to check if that already ran
 DIST_TEST=dist/manifest.json
 PACKAGE_NAME := $(shell awk '/"name":/ {gsub(/[",]/, "", $$2); print $$2}' package.json)
+RPM_NAME := $(PACKAGE_NAME)
+VERSION := $(shell T=$$(git describe 2>/dev/null) || T=1; echo $$T | tr '-' '.')
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply Makefile:22]: Let's see :)

**Comment by KKoukiou** [comment]
> I don't have more to add to Jirka. The only thing I didn't look at much is the `makeupdates` script which is already being discussed (and could be improved separately I guess)

@jkonecny12 @rvykydal shrinked it down to the bare minimum. I think all your makeupdates comments are now addresses.

File: packaging/anaconda-webui.spec.in [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+Release:        1%{?dist}
+Summary:        Anaconda installer Web interface
+License:        LGPL-2.1-or-later AND MIT
+URL:            https://github.com/rhinstaller/%{name}
+
+Source0:        https://github.com/rhinstaller/%{name}/releases/download/%{version}/%{name}-%{version}.tar.xz
+BuildArch:      noarch
+BuildRequires:  libappstream-glib
+BuildRequires:  make
+BuildRequires: gettext
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply packaging/anaconda-webui.spec.in]: Fixed

File: scripts/makeupdates (hunk 8 above)
[the remark this answers is not in the document]
Comment (KKoukiou) [reply scripts/makeupdates]: Fixed. Script was trimmed down to contain only used options.

File: scripts/makeupdates [hunk 15]
```diff
@@ -0,0 +1,174 @@
...
+
+def usage(cmd):
+    sys.stdout.write("Usage: %s [OPTION]...\n" % (cmd,))
+    sys.stdout.write("Options:\n")
+    sys.stdout.write("    -k, --keep       Do not delete updates subdirectory.\n")
+    sys.stdout.write("    -h, --help       Display this help and exit.\n")
+    sys.stdout.write("    -t, --tag        Make image from TAG to HEAD.\n")
+    sys.stdout.write("    -o, --offset     Make image from (latest_tag - OFFSET) to HEAD.\n")
+    sys.stdout.write("    -a, --add        Add contents of rpm to the update\n")
+    sys.stdout.write("    -p, --python     Version of python (for example, 3.9)\n")
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply scripts/makeupdates]: Honestly don't know, that copied from the original script so. Let's address in followup, all the code now in the script is used.

File: scripts/makeupdates [hunk 16]
```diff
@@ -0,0 +1,174 @@
...
+
+def main(argv):
+    prog = os.path.basename(argv[0])
+    cwd = os.getcwd()
+    spec = os.path.realpath(cwd + '/anaconda-webui.spec')
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply scripts/makeupdates:51]: We use this script after creating the RPMs that we are installaing in the VM. So always the spec file is there for our scenario.
Let's not over-optimize the script for scenarios we don't use it for currently, I want to remove this anyway. See jira linked in the commit message.

File: scripts/makeupdates (hunk 9 above)
[the remark this answers is not in the document]
Comment (KKoukiou) [reply scripts/makeupdates]: Fixed.

File: scripts/makeupdates [hunk 17]
```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python3
+#
+# makeupdates - Generate an updates.img containing changes since the last
+#               tag.
+#
+# Copyright (C) 2009-2013  Red Hat, Inc.
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply scripts/makeupdates]: Fixed.

File: .github/workflows/release.yml (hunk 2 above)
Comment (jkonecny12) [reply .github/workflows/release.yml:33]: Understand, would be nice but probably not requirement.

File: README.rst (hunk 5 above)
Comment (jkonecny12) [reply README.rst:25]: Interesting.

File: scripts/makeupdates (hunk 15 above)
[the remark this answers is not in the document]
Comment (KKoukiou) [reply scripts/makeupdates]: Removed.

**Review by rvykydal — APPROVED** [review]

**Review by M4rtinK — APPROVED** [review]
Looks good to me as far as I can tell. :)

**Comment by KKoukiou** [comment]
I will merge on monday early :)

File: Makefile [hunk 18]
```diff
@@ -18,7 +18,7 @@
 # one example file in dist/ to check if that already ran
 DIST_TEST=dist/manifest.json
 DIST_TEST_PRODUCTION=dist/index.css.gz
-PACKAGE_NAME=anaconda-webui
+PACKAGE_NAME := $(shell awk '/"name":/ {gsub(/[",]/, "", $$2); print $$2}' package.json)
```
Comment (VladimirSlavik) [inline Makefile]: Why is this needed? Will the package change names often?

**Review by VladimirSlavik — APPROVED** [review]
I mostly breezed through, but looks good to me. Thank you - this is a lot of work.

File: Makefile (hunk 18 above)
Comment (VladimirSlavik) [reply Makefile]: Ah, it's not in the final thing... ok...

File: .github/workflows/cockpit-lib-update.yml [hunk 19]
```diff
@@ -0,0 +1,31 @@
+name: cockpit-lib-update
```
Comment (VladimirSlavik) [inline .github/workflows/cockpit-lib-update.yml:1]: Do we want it? Will it work? Just making sure what's the status, no need to solve that now.

**Review by VladimirSlavik — APPROVED** [review]
LGTM!

File: .github/workflows/cockpit-lib-update.yml (hunk 19 above)
Comment (KKoukiou) [reply .github/workflows/cockpit-lib-update.yml:1]: Sorry I just saw this. I will make sure to test this today
